charon/src/ids/generator.rs

```diff
@@ -0,0 +1,59 @@
+use index_vec::Idx;
+use std::{collections::HashMap, hash::Hash, marker::PhantomData};
+
+#[derive(Debug, Clone, Copy)]
+pub struct Generator<I: Idx> {
+    counter: usize,
+    phantom: PhantomData<I>,
```

Comment:
Is this the standard way of doing (rather than storing an `I` and converting it to a `usize` when needs be)?

Reply:
I didn't think very hard, I think this is fine